### PR TITLE
Add RuleDefinitionsRepository

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Part of #4

### CODE REVIEW

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/packages.lock.json [hunk 1]
```diff
@@ -11,6 +11,18 @@
           "Microsoft.NETCore.Platforms": "1.1.0"
         }
       },
+      "Newtonsoft.Json": {
```
Comment (rita-sonarsource) [inline omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/packages.lock.json:14]: I uploaded Newtonsoft.Json and System.ComponentModel.Composition to the nuget feed, to use their netstandard 2.0 versions.
I also uploaded `Microsoft.NETCore.Platforms.1.1.0` -- it was a transitive dependency of `NETStandard.Library` and the build was failing without it.

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin.UnitTests/TestingInfrastructure/MefTestHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.ComponentModel.Composition.Hosting;
+using System.ComponentModel.Composition.Primitives;
+using System.Linq;
+using FluentAssertions;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests
+{
+    public static class MefTestHelpers
```
Comment (rita-sonarsource) [inline omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin.UnitTests/TestingInfrastructure/MefTestHelpers.cs:31]: Copied as-is from SLVS, along with the other 2 files (Exceptions + SingleObjectImporter), so we could use the Mef tests helper.

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin.UnitTests/packages.lock.json [hunk 3]
```diff
@@ -54,29 +54,6 @@
         "resolved": "16.6.1",
         "contentHash": "nBYXDgAZCfjsOVzlhMB5olGvX4dTDWB/gWaYS/MhgXBcCz8XJuVGqkfK8LmwlBR/eeUPE9Q/NFZNwlJyMZf0vg=="
       },
-      "Microsoft.CSharp": {
```
Comment (rita-sonarsource) [inline omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin.UnitTests/packages.lock.json:57]: I'm not sure where this dependency came from. The lock file was updated automatically.

**Comment by rita-sonarsource** [comment]
@duncanp-sonar please review, this is an initial commit.

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin.UnitTests/packages.lock.json (hunk 3 above)
Comment (duncanp-sonar) [reply omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin.UnitTests/packages.lock.json:57]: Probably from NewtonSoft - [see here](https://github.com/JamesNK/Newtonsoft.Json/issues/1270). If I remember correctly, some C# syntactic sugar keywords will cause the compiler to add a ref to that assembly (e.g. _dynamic_?)

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/SonarLint.OmniSharp.Plugin.csproj [hunk 4]
```diff
@@ -2,6 +2,12 @@
 
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
+    <LangVersion>8</LangVersion>
```
Comment (duncanp-sonar) [inline omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/SonarLint.OmniSharp.Plugin.csproj:5]: `netstandard2.0` defaults to `v7.3`. Why would we risk targeting a higher-level version of the language?

**Review by duncanp-sonar — COMMENTED** [review]
Looks ok, except for the question about setting the language version in the product code.

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/SonarLint.OmniSharp.Plugin.csproj (hunk 4 above)
Comment (rita-sonarsource) [reply omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/SonarLint.OmniSharp.Plugin.csproj:5]: As we discussed offline - just for synthetic sugar "using" statement.

**Pull request merged**